src/components/KCollapse/KCollapse.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <div class="k-collapse w-100">
+    <div
+      :class="{
+        'd-flex': triggerAlignment === 'trailing',
+        'd-block': triggerAlignment === 'leading'
+      }"
+      class="k-collapse-heading w-100 mb-3"
+    >
+      <div
+        v-if="title"
+        :class="{
+          'mr-auto': triggerAlignment === 'trailing',
+          'mb-2': triggerAlignment === 'leading'
+        }"
+        class="k-collapse-title"
+      >
+        {{ title }}
+      </div>
+      <div
+        :class="{
+          'ml-auto': triggerAlignment === 'trailing'
+        }"
+        class="k-collapse-trigger"
+      >
+        <slot
+          name="trigger"
+          :is-collapsed="getIsCollapsed()"
+          :toggle="toggleDisplay"
+        >
+          <div
+            class="k-collapse-trigger-content"
+            @click="toggleDisplay()"
+          >
+            <slot name="trigger-content">
+              <span v-if="triggerLabel">
+                <KIcon
+                  :icon="getIsCollapsed() ? 'chevronRight' : 'chevronDown'"
+                  color="var(--blue-500)"
+                  size="10"
+                  class="mr-1"
+                />
+                {{ triggerLabel }}
+              </span>
+              <KIcon
+                v-else
+                :icon="getIsCollapsed() ? 'chevronRight' : 'chevronDown'"
+                color="var(--black-70)"
+              />
+            </slot>
+          </div>
+        </slot>
+      </div>
+    </div>
+    <div
+      v-if="hasVisibleContent"
+      class="k-collapse-visible-content w-100"
+    >
+      <slot name="visible-content" />
+    </div>
+    <div
+      v-if="!getIsCollapsed()"
+      class="k-collapse-hidden-content w-100"
+    >
+      <slot />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ref, watch } from 'vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+
+export default defineComponent({
+  name: 'KCollapse',
+  components: {
+    KIcon,
+  },
+  props: {
+    modelValue: {
+      type: Boolean,
+      default: true,
+    },
+    title: {
+      type: String,
+      default: '',
+    },
+    triggerLabel: {
+      type: String,
+      default: '',
+    },
+    triggerAlignment: {
+      type: String,
+      default: 'trailing',
+      validator: (value: string): boolean => {
+        return ['leading', 'trailing'].includes(value)
```

Comment:
I am using leading / trailing in the event that we ever decide to support RTL languages. In general it is better to avoid left/right so that you can use the same prop value regardless of whether the browser is display LTR or RTL.

Reply:
Wouldn't left/right remain constant since it's not rendering paragraph text but rather referencing the left/right of the browser viewport? Was just a question, nothing to see here 😃

Reply:
Yeah, so, the way to look at it is this: we either are going to want to display the trigger in line with/opposite the title or below it. 

If we are in LTR language, all text is `left` aligned; so if we are displaying the trigger inline with the title we would want the trigger on the `right`, if we are displaying the trigger below the title, we would want it displayed on the `left`.

If we are in RTL language, all text is `right` aligned; so if we are displaying the trigger inline with the title we then switch to wanting the trigger on the `left`, and if we want to display the trigger below the title, to maintain vertical alignment we now want the trigger on the `right`.

By using `leading`/`trailing`, `leading` refers to `left` in LTR languages and `right` in RTL languages. `trailing` is the opposite, `right` in LTR languages and `left` in RTL languages. This means we don't have to pay attention to whether we are in LTR or RTL to correctly position the trigger. `leading` will always place us directly beneath the title and `trailing` will always place us opposite it.